test/MarketContractOraclize.js

```diff
@@ -180,15 +183,244 @@ contract('MarketContractOraclize', function(accounts) {
+    it("should only allow remaining quantity to be filled for an overfilled trade.", async function() {
+        const timeStamp = ((new Date()).getTime() / 1000) + 60*5; // order expires 5 minute from now.
+        const orderAddresses = [accountMaker, accountTaker, accounts[2]];
+        const unsignedOrderValues = [0, 0, entryOrderPrice, timeStamp, 1];
+        const orderQty = 5;   // user is attempting to buy 5
+        const qtyToFill = 10; // order is to be filled by 10
+        const orderHash = await orderLib.createOrderHash.call(
+            MarketContractOraclize.address,
+            orderAddresses,
+            unsignedOrderValues,
+            orderQty
+        );
+
+        const expectedQtyFilled = 5;
+
+        // Execute trade between maker and taker for overfilled amount of order.
+        const orderSignature = utility.signMessage(web3, accountMaker, orderHash)
+        const actualQtyFilled = await marketContract.tradeOrder.call(
+            orderAddresses,
+            unsignedOrderValues,
+            orderQty, // 5
+            qtyToFill, // 10
+            orderSignature[0],  // v
+            orderSignature[1],  // r
+            orderSignature[2],  // s
+            {from: accountTaker}
+        );
+
+        assert.equal(expectedQtyFilled, actualQtyFilled.toNumber(), "Quantity filled doesn't match expected");
+    })
+
+    it("should only allow remaining quantity to be cancelled for an over cancelled trade.", async function() {
+        const timeStamp = ((new Date()).getTime() / 1000) + 60*5; // order expires 5 minute from now.
+        const orderAddresses = [accountMaker, accountTaker, accounts[2]];
+        const unsignedOrderValues = [0, 0, entryOrderPrice, timeStamp, 1];
+        const orderQty = -5;   // user is attempting to sell 5
+        const qtyToFill = -1; // order is to be filled by 1
+        const orderHash = await orderLib.createOrderHash.call(
+            MarketContractOraclize.address,
+            orderAddresses,
+            unsignedOrderValues,
+            orderQty
+        );
+
+        // Execute trade between maker and taker for partial amount of order.
+        const orderSignature = utility.signMessage(web3, accountMaker, orderHash)
+        await marketContract.tradeOrder.call(
+            orderAddresses,
+            unsignedOrderValues,
+            orderQty, // -5
+            qtyToFill, // fill one slot
+            orderSignature[0],  // v
+            orderSignature[1],  // r
+            orderSignature[2],  // s
+            {from: accountTaker}
+        );
+
+        const expectedQtyCancelled = -5
+        const qtyToCancel = -7;
+        // over cancel order
+        const actualQtyCancelled = await marketContract.cancelOrder.call(
+            orderAddresses,
+            unsignedOrderValues,
+            orderQty,
+            qtyToCancel);
+
+        assert.equal(expectedQtyCancelled, actualQtyCancelled.toNumber(), "Quantity cancelled doesn't match expected.");
```

Comment:
@perfectmak I think there may be an issue here, I dont think with the test, but I think the underlying code if this test is passing.  If we have an order for qty of 5, and we trade 1, we should only then be able to cancel a max of 4 since that is the remaining qty.  You agree?  going to track this down

Reply:
I am sorry, I misunderstood what you where trying to accomplish.  I didn't realize you where using .call() on the tradeOrder function.  Therefore the state is not modified. Is this what you intended?

Reply:
Yes. Actually, the contract works. It is test that is wrong. I was focusing more on the cancelOrder's test and didn't notice that I used `.call()` to tradeOrder. I'll push a fix now.